### PR TITLE
[[ FFI ]] Add ability to convert proper list to c-array

### PR DESCRIPTION
**Pull request by livecodeali**

### CODE REVIEW

File: libfoundation/include/foundation.h [hunk 1]
```diff
@@ -3391,6 +3391,8 @@ MC_DLLEXPORT bool MCProperListCreate(const MCValueRef *values, uindex_t length,
 // foreign value is created (MCForeignValueRef).
 MC_DLLEXPORT bool MCProperListCreateWithForeignValues(MCTypeInfoRef type, const void *values, uindex_t value_count, MCProperListRef& r_list);
 
+MC_DLLEXPORT bool MCProperListConvertToForeignValues(MCProperListRef list, MCTypeInfoRef p_typeinfo, void*& r_values_ptr, uindex_t& r_values_count);
```
Comment (runrevmark) [inline libfoundation/include/foundation.h]: It is probably worth adding a description comment here like the converse function above.

**Comment by runrevmark** [comment]
@livecode-vulcan review ok c5d7735

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok c5d7735617be6d96e523658941a173fa6f9187c7

**Comment by livecodeali** [comment]
@livecode-vulcan retry c5d7735

**Comment by livecode-vulcan** [comment]
😎 test success c5d7735617be6d96e523658941a173fa6f9187c7

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1065)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1081)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1055)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/8)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/8)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1060)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1082)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1065)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1081)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1064)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1074)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1097)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1060)

**Pull request merged**